core/server/api/authentication.js

```diff
@@ -260,22 +272,21 @@ authentication = {
         });
     },
 
-    revoke: function (object) {
-        var token;
-
-        if (object.token_type_hint && object.token_type_hint === 'access_token') {
-            token = dataProvider.Accesstoken;
-        } else if (object.token_type_hint && object.token_type_hint === 'refresh_token') {
-            token = dataProvider.Refreshtoken;
-        } else {
-            return errors.BadRequestError('Invalid token_type_hint given.');
+    updateSetup: function updateSetup(object, options) {
+        if (!options.context || !options.context.user) {
+            return Promise.reject(new errors.NoPermissionError('You are not logged in.'));
         }
 
-        return token.destroyByToken({token: object.token}).then(function () {
-            return Promise.resolve({token: object.token});
-        }, function () {
-            // On error we still want a 200. See https://tools.ietf.org/html/rfc7009#page-5
-            return Promise.resolve({token: object.token, error: 'Invalid token provided'});
+        return dataProvider.User.findOne({role: 'Owner', status: 'all'}).then(function (result) {
+            var user = result.toJSON();
+
+            if (user.id !== options.context.user) {
+                return Promise.reject(new errors.NoPermissionError('You are not the blog owner.'));
+            }
```

Comment:
While I'm here, I've been wondering if we should do `canThis.user.edit.user()` here - seeing as we're already asserting that you must be the owner, it's probably not necessary.